### PR TITLE
[NativeAOT-LLVM] Include JS interop generator in ILC package

### PR DESCRIPTION
**Pull request by maraf**
- Include JS interop generator in ILC package
- Replace the one from TargetingPack
- Packages validated locally

Contributes to https://github.com/dotnet/runtimelab/issues/2434

### CODE REVIEW

File: src/coreclr/nativeaot/BuildIntegration/BuildIntegration.proj [hunk 1]
```diff
@@ -9,6 +9,11 @@
     <Content Include="*.*" Exclude="$(MSBuildProjectFile)" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)\System.Runtime.InteropServices\gen\Microsoft.Interop.SourceGeneration\Microsoft.Interop.SourceGeneration.csproj" OutputItemType="Analyzer" />
```
Comment (maraf) [inline src/coreclr/nativeaot/BuildIntegration/BuildIntegration.proj]: @akoeplinger How can I do something this in here? I need to be sure that the interop generator is built before this project's `Build` target

File: src/coreclr/nativeaot/BuildIntegration/BuildIntegration.proj [hunk 2]
```diff
@@ -9,7 +11,12 @@
     <Content Include="*.*" Exclude="$(MSBuildProjectFile)" />
   </ItemGroup>
 
-  <Target Name="Build">
+  <!-- TODO-LLVM: This is needed only until we ship Microsoft.NETCore.App.Ref containing the right version of generator -->
```
Comment (SingleAccretion) [inline src/coreclr/nativeaot/BuildIntegration/BuildIntegration.proj]:
```suggestion
  <!-- TODO-LLVM-Upstream: <new issue link>. -->
```
Could you open a new issue about this?

<details>
<summary>TODO-LLVM-Upstream?</summary>

We use this to tag places that either:
1) Represent concessions in code readability with the goal of conflict avoidance.
2) Represent technical debt (i. e. workarounds) due to the downstream status.

The difference with a plain `TODO-LLVM` is that `TODO-LLVM`s can be addressed downstream.
</details>

**Review by SingleAccretion — APPROVED** [review]
LGTM modulo a request for a new issue to link to. Thank you!

**Comment by akoeplinger** [comment]
LGTM

**Pull request merged**